pkg/reconciler/mtbroker/broker.go

```diff
@@ -156,13 +156,19 @@ func (r *Reconciler) reconcileKind(ctx context.Context, b *eventingv1.Broker) (*
 		// Ok to return nil for error here, once channel address becomes available, this will get requeued.
 		return &chanMan.ref, nil
 	}
-	if url := triggerChan.Status.Address.URL; url.Host == "" {
+	if url := triggerChan.Status.Address.URL; url == nil || url.Host == "" {
 		logging.FromContext(ctx).Debugw("Trigger Channel does not have an address", zap.Any("triggerChan", triggerChan))
 		b.Status.MarkTriggerChannelFailed("NoAddress", "Channel does not have an address.")
 		// Ok to return nil for error here, once channel address becomes available, this will get requeued.
 		return &chanMan.ref, nil
 	}
 
+	// Attach the channel address as a status annotation.
+	if b.Status.Annotations == nil {
+		b.Status.Annotations = make(map[string]string, 1)
+	}
+	b.Status.Annotations["channelAddress"] = triggerChan.Status.Address.URL.String()
```

Comment:
nit. I do wonder if this should be prefixed with knative.dev/channelAddress.
Since it's set by the controller there are no issues like with the meta where annotations could collide, so it's fine to go in as is but we might want to think about this before cutting the release.

Reply:
:+1: Seems like a good practice. Maybe in the future status.annotations would be patched by other controllers. I created #3626 to track this.